src/cell.h

```diff
@@ -300,20 +300,27 @@ struct Cell {
         doc->AddUndo(this);
     }
 
-    void Save(wxDataOutputStream &dos) const {
+    void Save(wxDataOutputStream &dos, Cell *ocs) const {
         dos.Write8(celltype);
         dos.Write32(cellcolor);
         dos.Write32(textcolor);
         dos.Write8(drawstyle);
+        uint cellflags = 0x0;
         if (HasTextState()) {
-            dos.Write8(grid ? TS_BOTH : TS_TEXT);
+            cellflags = grid ? TS_BOTH : TS_TEXT;
+            if(this == ocs) cellflags |= TS_SELECTION_MASK;
+            dos.Write8(cellflags);
             text.Save(dos);
-            if (grid) grid->Save(dos);
+            if (grid) grid->Save(dos, ocs);
         } else if (grid) {
-            dos.Write8(TS_GRID);
-            grid->Save(dos);
+            cellflags = TS_GRID;
+            if(this == ocs) cellflags |= TS_SELECTION_MASK;
+            dos.Write8(cellflags);
+            grid->Save(dos, ocs);
         } else {
-            dos.Write8(TS_NEITHER);
+            cellflags = TS_NEITHER;
+            if(this == ocs) cellflags |= TS_SELECTION_MASK;

```

Comment:
why is this done for each branch of `if` ? please move all code thats the same out of the `if`

Reply:
Thanks for your feedback. I revised this. Is it ok now?